src/components/ContactDetailsModal/ContactDetailsModal.js

```diff
@@ -0,0 +1,275 @@
+`;
+
+export const LoadingSpinner = styled(Spinner)`
+  margin-top: ${spacing.large}px;
+  align-items: center;
+  justify-content: center;
+`;
+
+export const TitleWrapper = styled.View`
+  flex-direction: row;
+  justify-content: center;
+  align-items: center;
+  width: 100%;
+`;
+
+export const QRCodeButton = styled.TouchableOpacity`
+  position: absolute;
+  top: 0;
+  right: 0;
+`;
+
+
+export const QRCodeIcon = styled(Icon)`
+  color: ${({ color }) => color};
+  font-size: 20px;
+`;
+
+const renderContactInput = (
+  value: string,
+  onChangeText: (value: string) => void,
+  placeholder: string,
+  icon: string,
+  theme: Theme,
+) => (
+  <InputWrapper>
+    <FieldIcon source={icon} />
+    <TextInput
+      theme={theme}
+      inputWrapperStyle={{ flex: 1, paddingBottom: 0 }}
+      inputProps={{
+        value,
+        onChangeText,
+        placeholder,
+      }}
+    />
+  </InputWrapper>
+);
+
+const ContactDetailsModal = ({
+  isVisible,
+  onModalHide,
+  theme,
+  contact,
+  onSavePress,
+  isDefaultNameEns,
+  title,
+  contacts,
+  showQRScanner,
+}: Props) => {
+  const [addressValue, setAddressValue] = useState('');
+  const [nameValue, setNameValue] = useState('');
+  const [dirtyInputs, setDirtyInputs] = useState(false);
+  const [resolvingEns, setResolvingEns] = useState(false);
+  const [isScannerVisible, setIsScannerVisible] = useState(false);
+  const [ensUnresolved, setEnsUnresolved] = useState(false);
+  const { walletIcon, personIcon } = images(theme);
+
+  // reset input value on default change
+  useEffect(() => {
+    setDirtyInputs(false);
+    setAddressValue(contact?.ethAddress || '');
+    const defaultName = isDefaultNameEns ? contact?.ensName : contact?.name;
+    setNameValue(defaultName || '');
+  }, [contact]);
+
+  useEffect(() => {
+    if (!dirtyInputs && (!isEmpty(nameValue) || !isEmpty(addressValue))) {
+      setDirtyInputs(true);
+    }
+    if (ensUnresolved) setEnsUnresolved(false); // reset
+  }, [nameValue, addressValue]);
+
+  useEffect(() => {
+    if (resolvingEns
+      || !isEmpty(nameValue)
+      || !isValidAddress(addressValue)
+      || isEnsName(addressValue)) return;
+
+    setResolvingEns(true);
+    lookupAddress(addressValue)
+      .then((ensName) => {
+        if (ensName) setNameValue(ensName);
+        setResolvingEns(false);
+      })
+      .catch(() => setResolvingEns(false));
+  }, [addressValue]);
+
+  useEffect(() => {
+    if (resolvingEns
+      || !isEmpty(addressValue)
+      || !isEnsName(nameValue)) return;
+
+    setEnsUnresolved(false);
+    setResolvingEns(true);
+    getReceiverWithEnsName(nameValue)
+      .then(({ receiver }) => {
+        if (receiver) {
+          setAddressValue(receiver);
+        } else {
+          setEnsUnresolved(true);
+        }
+        setResolvingEns(false);
+      })
+      .catch(() => {
+        setEnsUnresolved(true);
+        setResolvingEns(false);
+      });
+  }, [nameValue]);
+
+  let errorMessage;
```

Comment:
User is allowed to add multiple contacts with same name, is that okay?

Reply:
Not sure, it wasn't described in AC, neither it was described for unique address check, but can add for names as well 👍